hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -160,9 +163,9 @@
    */
   private final List<WALEntryFilter> baseFilterOutWALEntries;
 
-  ReplicationSource() {
+  ReplicationSource(WALProvider walProvider) {
```

Comment:
I think the design here, is to not have any parameters for constructor, and then use init method below to do the initialize work? Otherwise if we could even pass a WALProvider here, why not also pass other parameters at the same time? At least, Configuration, FileSystem, Server will not be changed and easy to get when constructing a ReplicationSource.

Reply:
init is fat as it is but this is added on Constructor because factory does construction and it knows all WALProviders and which to pass. The init is done after Construction. The context on what WALProvider to supply is not present at init time.